feat: Add Anthropic Claude and LM Studio provider support

csfet9 opened the pull request:
## Summary
Adds support for Anthropic Claude and LM Studio as LLM providers for Hindsight.

## Changes
- **Anthropic Provider**: Full async support for Claude models (Sonnet 4, Haiku 4.5, etc.)
- **LM Studio Provider**: Support for local model inference via LM Studio's OpenAI-compatible API
- **JSON Compatibility Fix**: Handle markdown-wrapped JSON responses from local models
- **Documentation**: Updated `.env.example` with configuration examples for all providers

## Tested With
- ✅ Claude Sonnet 4 (`claude-sonnet-4-20250514`)
- ✅ Claude Haiku 4.5 (`claude-haiku-4-5-20251001`)
- ✅ Qwen 30B via LM Studio

## Example Configuration
```bash
# Anthropic
HINDSIGHT_API_LLM_PROVIDER=anthropic
HINDSIGHT_API_LLM_API_KEY=your-anthropic-key
HINDSIGHT_API_LLM_MODEL=claude-haiku-4-5-20251001

# LM Studio
HINDSIGHT_API_LLM_PROVIDER=lmstudio
HINDSIGHT_API_LLM_API_KEY=lmstudio
HINDSIGHT_API_LLM_BASE_URL=http://localhost:1234/v1
HINDSIGHT_API_LLM_MODEL=your-model-name
```

## Files Changed
- `hindsight-api/pyproject.toml` - Added anthropic dependency
- `hindsight-api/hindsight_api/engine/llm_wrapper.py` - Anthropic + LM Studio implementation
- `hindsight-api/hindsight_api/config.py` - LM Studio default URL
- `hindsight/hindsight/server.py` - Updated docstrings
- `.env.example` - Configuration examples

nicoloboschi reviewed (commented):
Thanks for your contribution! I've left 2 small comments

nicoloboschi commented on hindsight-api/hindsight_api/engine/llm_wrapper.py:
```diff
@@ -224,11 +271,31 @@ async def call(
-                        json_data = json.loads(content)
+                        
+                        # For local models, they may wrap JSON in markdown code blocks
+                        if self.provider in ("lmstudio", "ollama"):
```
this is a good thing, can we do it for all the providers? 
we don't enforce the schema checks because some models are too stupid and couldn't get it right all the time.

nicoloboschi commented on hindsight-api/hindsight_api/engine/llm_wrapper.py:
```diff
@@ -301,6 +368,144 @@ async def call(
+                    if block.type == "text":
+                        content += block.text
+
+                if response_format is not None:
```
let's create a private function instead of duplicating code

nicoloboschi commented on test_internal.py:
```diff
@@ -0,0 +1,38 @@
+import asyncio
```
remove?

csfet9 commented:
Thanks for the review! I've addressed all the feedback:                                                                             
                                                                                                                                      
  Changes made:                                                                                                                       
  Moved LLM_MAX_CONCURRENT to config.py with proper HINDSIGHT_API_ prefix                                                        
  Added HINDSIGHT_API_LLM_TIMEOUT config (removed the fragile model-size heuristic)                                              
  Applied markdown JSON extraction to all providers (not just local)                                                             
  Fixed a bug in Anthropic's markdown extraction (was missing .split("```")[0])                                                  
  Changed logger.info to logger.debug for LLM request/response logs                                                              
  Removed the local dev docker-compose.yml                                                                                       
                                                                                                                                      
  Testing note: Hindsight is working well with qwen/qwen3-vl-8b via LM Studio.
